### ISSUE TITLE
Bug: prefer-object-has-own autofix changes behavior when Object is shadowed

### ISSUE THREAD

**Issue by cchaeyoung** [eslint/eslint#21279]
### Environment

Node version: v22.19.0
npm version: v10.5.0
Local ESLint version: v10.9.0
Global ESLint version: Not found
Operating System: win32 10.0.26200

### What parser are you using?

Default (Espree)

### What did you do?

<details>
<summary>Configuration</summary>

```js
export default [
	{
		languageOptions: {
			sourceType: "script"
		},
		rules: {
			"prefer-object-has-own": "error"
		}
	}
];
```
</details>

```js
let Object = {
	prototype: {
		hasOwnProperty: {
			call() {}
		}
	}
};

Object.prototype.hasOwnProperty.call({}, "foo");
```

### What did you expect to happen?

No error should be reported and no autofix should be applied, because `Object` here refers to a user-defined global variable, not the built-in `Object` constructor.

### What actually happened?

```
Use 'Object.hasOwn()' instead of 'Object.prototype.hasOwnProperty.call()'.
```

With `--fix`, the last line is rewritten to

```js
Object.hasOwn({}, "foo");
```

### Link to Minimal Reproducible Example

https://eslint.org/play/#eyJ0ZXh0IjoibGV0IE9iamVjdCA9IHtcblx0cHJvdG90eXBlOiB7XG5cdFx0aGFzT3duUHJvcGVydHk6IHtcblx0XHRcdGNhbGwoKSB7fVxuXHRcdH1cblx0fVxufTtcblxuT2JqZWN0LnByb3RvdHlwZS5oYXNPd25Qcm9wZXJ0eS5jYWxsKHt9LCBcImZvb1wiKTtcbiIsIm9wdGlvbnMiOnsicnVsZXMiOnsicHJlZmVyLW9iamVjdC1oYXMtb3duIjpbImVycm9yIl19LCJsYW5ndWFnZU9wdGlvbnMiOnsicGFyc2VyT3B0aW9ucyI6eyJlY21hRmVhdHVyZXMiOnt9fSwic291cmNlVHlwZSI6InNjcmlwdCJ9fX0=

### Participation

- [x] I am willing to submit a pull request for this issue.

### AI acknowledgment

- [x] I did not use AI to generate this issue report.
- [ ] (If the above is not checked) I have reviewed the AI-generated content before submitting.

### Additional comments

Disclosure: I'm a participant of [open source contribution program OSSCA](https://github.com/eslint-ossca)

**Comment by DMartens** [comment]
Thank you for the issue. I am able to reproduce this when setting the source type to `script`.
The rule only checks that `Object` refers to a global in the scope but not that is no "defs".
A PR for this is welcome.
